codejar.ts

```diff
@@ -275,8 +283,8 @@ export function CodeJar(editor: HTMLElement, highlight: (e: HTMLElement, pos?: P
         legacyNewLineFix(event)
       }
 
-      // Place adjacent "}" on next line
-      if (newLinePadding !== padding && after[0] === '}') {
+      // Place adjacent "}" (or matching closing char) on next line
+      if (newLinePadding !== padding && reverseRegex(options.indentOn).test(after)) {
```

Comment:
Let’s introduce another regexp instead of trying to modify existing one. For example we can add an map with list of what brackets should be idented.

Reply:
Hum, if you have a look at my code, that is exactly what I have done.

Reply:
Let’s deke reverseRegexp func. It’s error prone.

Reply:
How is it error-prone? This function does not modify the existing regex but returns a new one. Just look at the code:

```js
function reverseRegex(regex: RegExp) {
    const r = regex.toString().replace(/\/\$/g, '');
    const chars = /^\[.+\]$/.test(r) ? r.slice(1, -1).split('') : [r];
    const open = `([{'"`;
    const close = `)]}'"`;
    return new RegExp('^' + chars.map((c) => close[open.indexOf(c)]).filter(Boolean).join(''));
  }
```

Unless you have a better option... But if you don't want people improving your code, just don't make it open-source and keep it buggy. That's your call.

Reply:
I mean instead of relying of format of  indentOn regexp we can introduce new config. What if regexp is /start/?

Reply:
Or /*+=/

Reply:
If the `indentOn` regex contains non-matching characters, then `reverseRegex()` returns a catch-all regex (`/^/`). No need to introduce another option (though this `indentOn` regex stuff seems inherently flawed).

But that's your library, so do as you please. As for myself, I am considering switching to CodeMirror for my project.

Reply:
Sound good.